tests/Unit/ControlSystem/Test_Controller.cpp

```diff
@@ -0,0 +1,223 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <cmath>
+
+#include "ControlSystem/Controller.hpp"
+#include "ControlSystem/ControllerPND.hpp"
+#include "ControlSystem/FunctionOfTime.hpp"
+#include "ControlSystem/PiecewisePolynomial.hpp"
+#include "ControlSystem/TimescaleTuner.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "ErrorHandling/Error.hpp"
+
+SPECTRE_TEST_CASE("Unit.ControlSystem.Controllers.PND",
+                  "[ControlSystem][Unit]") {
+  const double decrease_timescale_threshold = 1.0e-2;
+  const double increase_timescale_threshold = 1.0e-4;
+  const double increase_factor = 1.01;
+  const double decrease_factor = 0.99;
+  const double max_timescale = 10.0;
+  const double min_timescale = 1.0e-3;
+  const double initial_timescale = 1.0e-2;
+
+  TimescaleTuner tst({initial_timescale}, max_timescale, min_timescale,
+                     decrease_timescale_threshold, increase_timescale_threshold,
+                     increase_factor, decrease_factor);
+
+  // test following a sinusoidal target function
+  double t = 0.0;
+  const double dt = 1.0e-3;
+  const double final_time = 5.0;
+  const constexpr size_t deriv_order = 2;
+  const double freq = 3.0;
+
+  // properly initialize the function of time to match our target function
+  const std::array<DataVector, deriv_order + 1> init_func{
+      {{std::sin(freq * t)},
+       {freq * std::cos(freq * t)},
+       {-square(freq) * std::sin(freq * t)}}};
+  FunctionsOfTime::PiecewisePolynomial<deriv_order> f_of_t_derived(t,
+                                                                   init_func);
```

Comment:
Can `init_func` be inlined? I think you can also make it non-const and `std::move` it into `f_of_t_derived` - I'm also making these comments for my own learning purposes too so can someone else weigh in on whether I'm advising the right thing here?